src/main/java/org/jspecify/nullness/NullMarked.java

```diff
@@ -73,13 +73,13 @@
  *             type argument in {@code MyList<@Nullable Foo>} is out-of-bounds, so the list elements
  *             are always non-null. (<a href="https://bit.ly/3ppb8ZC">Why?</a>)
  *       </ul>
- *   <li>A usage of a <b>nullable-bounded type variable</b> (such as the {@code E} in {@code class
- *       Foo<E extends @Nullable Bar>}) is not considered nullable, non-null, or even of
- *       "unspecified" nullness. Rather it has <b>parametric nullness</b>. In order to support both
- *       nullable and non-null type arguments safely, the {@code E} type itself must be handled
- *       <i>strictly</i>: as if nullable when "read from", but as if non-null when "written to".
- *       (Contrast with {@code class Foo<E extends Bar>}, where usages of {@code E} are simply
- *       non-null, just like usages of {@code String}.)
+ *   <li>When a type variable has nullable upper bound, such as the {@code E} in {@code class Foo<E
+ *       extends @Nullable Bar>}), an unannotated usage of this type variable is not considered
+ *       nullable, non-null, or even of "unspecified" nullness. Rather it has <b>parametric
+ *       nullness</b>. In order to support both nullable and non-null type arguments safely, the
```

Comment:
wdyt about using the notion of "type variance" here instead of parametricity? It feels closer to me to what we're talking about. When I hear parametric I think "can be passed as argument" or at least, "determined by argument" which is not really true here, based on what the text says.

Reply:
this'll have to be a broader discussion than just this PR.